### PR TITLE
Action group button component module

### PR DESCRIPTION
**Pull request by ramirezStefano**
Cleaned up the provider file that had unused code. Module ready for final review!

### CODE REVIEW

File: src/components/ActionGroupButtonComponent/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
+## Notes about the component
+
+OK so some notes on the made component the process and the use of the stroy book and how everything is wired up.
```
Comment (eduardo-vargas) [inline src/components/ActionGroupButtonComponent/README.md]:
```suggestion
OK so some notes on the component, the process and the use of the storybook and how everything is wired up.
```

File: src/components/ActionGroupButtonComponent/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+## UI
+
+First the integration of the UI. This was done though the return statement of the FC where all the html tags and JSX components are, here the pertinent keys that are used for such elements are defined and thus be used in the component behaviour logic. e.g. 'isBold' key in the react spectrum components.
+
+## Pertinent Properties
+
+First once the user interface has been defined the needed interface values need to be also defined, with the relevant types, the _?_ if the value is to be optional, if it is left without asignment its undefined. as such:
```
Comment (eduardo-vargas) [inline src/components/ActionGroupButtonComponent/README.md]:
```suggestion
First once the user interface has been defined the needed interface values need to be also defined, with the relevant types, the `?` if the value is to be optional, if it is left without assignment its undefined. as such:
```

File: src/components/ActionGroupButtonComponent/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+This are the components to be passed both in the FC:
+
+```typescript
+const ActionGroupButtonComponent: FC<Props> = ({ isBold, isItalics }) => {};
+```
+
+that are then accessible in the storybook expanded arguments.
+
+```typescript
+[...args];
```
Comment (eduardo-vargas) [inline src/components/ActionGroupButtonComponent/README.md]:
```suggestion
{...args};
```

File: src/components/ActionGroupButtonComponent/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+### Setting the values.
+
+The initial request was to use an object yet as the problem, but after some refinement instead of an object the next best implementation was for the use of given useState hook provided by the Adobe documentation:
+
+```typescript
+const [selected, setSelected] = React.useState<Selection>(new Set(["list"]));
+```
+
+Then due to the fact that we needed to set the interface values, and this was not done by the default hook the implementation of an complex hook was implemented, with the explicit use of the anonimous funtion, like so:
```
Comment (eduardo-vargas) [inline src/components/ActionGroupButtonComponent/README.md]:
```suggestion
Then due to the fact that we needed to set the interface values, and this was not done by the default hook, a complex hook was implemented, with the explicit use of the anonymous function, like so:
```

File: src/components/ActionGroupButtonComponent/README.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+if (isItalics) {
+    initialValue.push("italics");
+}
+
+return new Set(initialValue);
+```
+
+All of this within the complex useState hook.
+
+Very important to return the expected type, in this case Selection expects a Set so a new Set is made with the elements of the local array that have been pushed.
```
Comment (eduardo-vargas) [inline src/components/ActionGroupButtonComponent/README.md]:
```suggestion
Very important to return the expected type, in this case Selection expects a `Set` so a new Set is created with the local array that has been modified within the anonymous function.
```

File: src/components/ActionGroupButtonComponent/README.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+At this point the setSelected is called every time onSelectionChange prop is called this updating the values of the array. And being tracked by the selectedKeys prop of the Action Group component. With this impementation we have controlled by having the selectedKeys prop having the selected interface values.
```
Comment (eduardo-vargas) [inline src/components/ActionGroupButtonComponent/README.md]:
```suggestion
At this point the setSelected is called every time `onSelectionChange` prop is called thus updating the values of the array. And being tracked by the `selectedKeys` prop of the Action Group component. With this implementation we have a controlled  ActionGroup by passing the `selectedKeys` prop with the selected values.
```

File: src/components/ActionGroupButtonComponent/README.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+## Storybook setup
+
+Within the storybook the props from the interface are passed with the ...args, thus the Storybook has access to the interface references that have been also passe into the FC.
```
Comment (eduardo-vargas) [inline src/components/ActionGroupButtonComponent/README.md]:
```suggestion
Within the storybook the props from the interface are passed with the `...args`, thus the Storybook has access to the props/arguments that have been also passed into the Component.
```

File: src/components/ActionGroupButtonComponent/README.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+We then make the specific stories that are all the branches of possible state combinations in this case, all switches off, all on and one or the other. this follows the following syntax e.g.:
```
Comment (eduardo-vargas) [inline src/components/ActionGroupButtonComponent/README.md]:
```suggestion
We then make the specific stories that are all the bound to the sample story template (using the `bind()` method). Each story illustrates a different possible state, in this case,
1. on and on
2. off and off
3. on and off  
4. off and on

 This follows the following syntax e.g.:
```

**Review by eduardo-vargas — APPROVED** [review]
Looks good! Some comments on the documentation

**Comment by ramirezStefano** [comment]
Will keep an eye out future README typos and double check the content.

**Pull request merged**